components/authz-service/storage/postgres/migration/sql/54_use_project_db_id_in_references_of_project_rules.up.sql

```diff
@@ -0,0 +1,313 @@
+  INNER JOIN iam_rule_conditions AS rc
+  ON rc.rule_db_id=r.db_id
+  INNER JOIN iam_projects AS p
+  ON r.project_id=p.db_id
+  WHERE projects_match_for_rule(p.id, _project_filter)
+  GROUP BY r.id, p.id, r.name, r.type;
+
+$$ LANGUAGE sql;
+
+CREATE OR REPLACE FUNCTION
+  query_rules(_project_filter TEXT[])
+  RETURNS setof json AS $$
+
+  SELECT json_build_object(
+    'id', r.id,
+    'project_id', p.id,
+    'name', r.name,
+    'type', r.type,
+    'status', 'applied',
+    'conditions', json_agg(
+      json_build_object(
+        'value', rc.value,
+        'operator', rc.operator,
+        'attribute', rc.attribute
+      )
+    )
+  ) AS rule
+  FROM iam_project_rules AS r
+  INNER JOIN iam_rule_conditions AS rc
+  ON rc.rule_db_id=r.db_id
+  INNER JOIN iam_projects AS p
+  ON r.project_id=p.db_id
+  WHERE projects_match_for_rule(p.id, _project_filter)
+  GROUP BY r.id, p.id, r.name, r.type;
+
+$$ LANGUAGE sql;
+
+CREATE OR REPLACE FUNCTION
+  query_staged_or_applied_rule(_rule_id TEXT, _project_filter TEXT[])
+  RETURNS json AS $$
+
+  -- check for applied rule
+  SELECT json_build_object(
+    'id', r.id,
+    'project_id', p.id,
+    'name', r.name,
+    'type', r.type,
+    'deleted', r.deleted,
+    'status', 'staged',
+    'conditions', json_agg(
+      json_build_object(
+        'value', rc.value,
+        'operator', rc.operator,
+        'attribute', rc.attribute
+      )
+    )
+  ) AS rule
+  FROM iam_staged_project_rules AS r
+  INNER JOIN iam_staged_rule_conditions AS rc
+  ON rc.rule_db_id=r.db_id
+  INNER JOIN iam_projects AS p
+  ON r.project_id=p.db_id
+  WHERE r.id=_rule_id AND projects_match_for_rule(p.id, _project_filter)
+  GROUP BY r.id, p.id, r.name, r.type, r.deleted
+
+  UNION ALL
+
+  -- check for applied rule
+  SELECT json_build_object(
+    'id', r.id,
+    'project_id', p.id,
+    'name', r.name,
+    'type', r.type,
+    'deleted', false,
+    'status', 'applied',
+    'conditions', json_agg(
+      json_build_object(
+        'value', rc.value,
+        'operator', rc.operator,
+        'attribute', rc.attribute
+      )
+    )
+  ) AS rule
+  FROM iam_project_rules AS r
+  INNER JOIN iam_rule_conditions AS rc
+   ON rc.rule_db_id=r.db_id
+  INNER JOIN iam_projects AS p
+  ON r.project_id=p.db_id
+  WHERE r.id=_rule_id AND projects_match_for_rule(p.id, _project_filter)
+  GROUP BY r.id, p.id, r.name, r.type;
+
+$$ LANGUAGE sql;
+
+-- NEW: it takes a project's db_id as first argument
+-- This way, we can keep some of the other functions unchanged.
+CREATE OR REPLACE FUNCTION
+  projects_match_for_rule(_project_db_id INTEGER, _projects_filter TEXT[])
+  RETURNS BOOLEAN AS $$
+    SELECT array_length(_projects_filter, 1) IS NULL
+    OR EXISTS (SELECT 1 FROM iam_projects WHERE db_id=_project_db_id AND id=ANY(_projects_filter))
+$$ LANGUAGE SQL;
+
+CREATE OR REPLACE FUNCTION update_rule(_in_rule_id text, _in_project_id text, _in_name text, _in_type text, projects_filter TEXT[])
```

Comment:
ℹ️ Only mildly adjusted. Besides sprinkling a few `project_db_id(_in_project_id)` over this, I've replaced
```
EXISTS (SELECT x, y FROM ...)
```
with
```
EXISTS (SELECT 1 FROM ...)
```
since I believe the fields we select don't matter at all there.